### PR TITLE
203 add sound in r type game

### PR DESCRIPTION
**Pull request by TomDUVAL-MAHE**
Add Sound Ressource and Sound Component

### CODE REVIEW

File: libs/GraphicECS/SFML/Components/SoundName.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+#ifndef SOUNDNAME_HPP_
+#define SOUNDNAME_HPP_
+
+#include "Component/Component.hpp"
+#include "GraphicECS/SFML/Resources/SoundResource.hpp"
+
+namespace ecs
+{
+    /// @brief SoundName store the key as enum to the corresponding sound stores in shared resource SoundResource.
```
Comment (osvegn) [inline libs/GraphicECS/SFML/Components/SoundName.hpp:16]: stores

File: libs/GraphicECS/SFML/Components/SoundName.hpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    class SoundName : public Component {
+      public:
+        /// @brief The sound name stores to find the corresponding soundName.
```
Comment (osvegn) [inline libs/GraphicECS/SFML/Components/SoundName.hpp:19]: The sound name stores an enum to find which sound it needs in SoundResources shared resource

File: tests/unit_tests/libs_tests/SFML_tests/SoundComponentTests.cpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    SoundName sound;
+
+    cr_assert_eq(0, sound.soundName);
+}
+
+Test(SoundName, create_sound_shoot)
+{
+    SoundName sound(SoundResource::SHOOT);
+
+    cr_assert_eq(1, sound.soundName);
```
Comment (osvegn) [inline tests/unit_tests/libs_tests/SFML_tests/SoundComponentTests.cpp:25]: Maybe you can compare directly with the enumerator

File: conanfile.txt [hunk 4]
```diff
@@ -1,7 +1,7 @@
 [requires]
 sfml/2.5.1
-boost/1.80.0
-zlib/1.2.13
+boost/1.79.0
```
Comment (LucasTesnier) [inline conanfile.txt:3]: Why a downgrade ?

File: conanfile.txt [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
+zlib/1.2.12
```
Comment (LucasTesnier) [inline conanfile.txt:4]: Why a downgrade to ?

File: libs/GraphicECS/SFML/Components/SoundName.hpp [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+#ifndef SOUNDNAME_HPP_
+#define SOUNDNAME_HPP_
+
+#include "Component/Component.hpp"
+#include "GraphicECS/SFML/Resources/SoundResource.hpp"
+
+namespace ecs
+{
+
```
Comment (LucasTesnier) [inline libs/GraphicECS/SFML/Components/SoundName.hpp:16]: useless empty line

File: libs/GraphicECS/SFML/Components/SoundName.hpp [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        inline void setStatus(status_e status) { _status = status; };
+
+        /// @brief This function return the name of the Sound
+        inline SoundResource::soundName_e getName() const { return soundName; };
+
+        /// @brief This function set the name of the Sound
+        inline void setName(SoundResource::soundName_e name) { soundName = name; };
+
+      private:
+        status_e _status;
```
Comment (LucasTesnier) [inline libs/GraphicECS/SFML/Components/SoundName.hpp:50]: A component should have all his variable in public

File: libs/GraphicECS/SFML/Components/SoundName.hpp [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        /// on the sf::Sound load in shared resource Sound.
+        SoundName(
+            SoundResource::soundName_e newSoundName = SoundResource::UNDEFINED, status_e status = status_e::STOPED)
+            : soundName(newSoundName), _status(status){};
+
+        /// @brief Default destructor of SoundName component.
+        ~SoundName() = default;
+
+        /// @brief This function return the status of the Sound
+        inline status_e getStatus() const { return _status; };
```
Comment (LucasTesnier) [inline libs/GraphicECS/SFML/Components/SoundName.hpp:38]: No need to do getter and setter, component have all his variable in public

File: libs/GraphicECS/SFML/Resources/SoundResource.cpp [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+
+using namespace ecs;
+
+void SoundResource::addSound(const soundName_e sound_e, const std::filesystem::path &soundPath)
+{
+    sf::SoundBuffer buffer;
+    sf::Sound sound;
+
+    if (!buffer.loadFromFile(soundPath.string()))
+        // throw error
```
Comment (LucasTesnier) [inline libs/GraphicECS/SFML/Resources/SoundResource.cpp:18]: Throw an error now ;) (Std::logic_error, or a custom one inside the ErrorLib)

File: libs/GraphicECS/SFML/Resources/SoundResource.hpp [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        /// @brief Enumeration of all available Sounds
+        enum soundName_e { UNDEFINED, SHOOT, DEATH, EXPLOSION };
+
+        /// @brief Name of map which contains Sounds.
+        using SoundsList = std::unordered_map<soundName_e, std::shared_ptr<sf::Sound>>;
+
+        /// @brief Add a Sound from it's Sound Path passed as parameter
+        /// @param sound_e Enum of the Sound
+        /// @param soundPath The sound path to be used.
+        inline SoundResource(const soundName_e sound_e, const std::filesystem::path &soundPath)
```
Comment (LucasTesnier) [inline libs/GraphicECS/SFML/Resources/SoundResource.hpp:30]: Can we directly passed a list of sound ? And why not ?

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Some questions and component error

File: libs/GraphicECS/SFML/Resources/SoundResource.hpp (hunk 10 above)
Comment (TomDUVAL-MAHE) [reply libs/GraphicECS/SFML/Resources/SoundResource.hpp:30]: Nop the buffer in sfml cannot support a multiple load

File: libs/GraphicECS/SFML/Components/SoundName.hpp [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+** SoundName
+*/
+
+#ifndef SOUNDNAME_HPP_
+#define SOUNDNAME_HPP_
+
+#include "Component/Component.hpp"
+#include "GraphicECS/SFML/Resources/SoundResource.hpp"
+
+namespace ecs
```
Comment (osvegn) [inline libs/GraphicECS/SFML/Components/SoundName.hpp:14]: Bad namespace, use graphicECS::SFML::Components

File: libs/GraphicECS/SFML/Resources/SoundResource.hpp [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+
+#ifndef SOUNDRESOURCES_HPP_
+#define SOUNDRESOURCES_HPP_
+
+#include <SFML/Audio.hpp>
+#include <filesystem>
+#include "Resource/Resource.hpp"
+#include <unordered_map>
+
+namespace ecs
```
Comment (osvegn) [inline libs/GraphicECS/SFML/Resources/SoundResource.hpp:16]: bad namespace, use graphicECS::SFML::Resources

**Review by osvegn — APPROVED** [review]
Bad namespaces, otherwise looks good

File: libs/GraphicECS/SFML/Components/SoundName.hpp (hunk 11 above)
Comment (TomDUVAL-MAHE) [reply libs/GraphicECS/SFML/Components/SoundName.hpp:14]: improve in #205 and #206 but yes i change that this afternoon

File: libs/GraphicECS/SFML/Resources/SoundResource.hpp (hunk 12 above)
Comment (TomDUVAL-MAHE) [reply libs/GraphicECS/SFML/Resources/SoundResource.hpp:16]: improve in #205 and #206 but yes i change that this afternoon

**Review by LucasTesnier — DISMISSED** [review]

**Review by osvegn — APPROVED** [review]

**Review by LucasTesnier — APPROVED** [review]

**Pull request merged**